server/lib/scheduler/Scheduler.ts

```diff
@@ -122,102 +127,67 @@ class Scheduler {
     }
   }
 
-  public async assignNextWorkspace(userId, maybeSuboptimal = false) {
+  public async assignNextWorkspace(userId: string): Promise<string | null> {
     this.resetCaches();
     this.schedule.leaveCurrentWorkspace(userId);
 
-    let actionableWorkspaces;
-    if (this.isInOracleMode.getValue()) {
-      actionableWorkspaces = await this.getActionableWorkspacesInOracleMode({
-        maybeSuboptimal: false,
-        userId,
-      });
-    } else {
-      actionableWorkspaces = await this.getActionableWorkspaces({
-        maybeSuboptimal: false,
-        userId,
-      });
-    }
-
-    if (actionableWorkspaces.length === 0 && maybeSuboptimal) {
-      if (this.isInOracleMode.getValue()) {
-        actionableWorkspaces = await this.getActionableWorkspacesInOracleMode({
-          maybeSuboptimal: true,
-          userId,
-        });
-      } else {
-        actionableWorkspaces = await this.getActionableWorkspaces({
-          maybeSuboptimal: true,
-          userId,
-        });
-      }
-    }
-
-    if (actionableWorkspaces.length === 0) {
-      const fallbackScheduler = await this.getFallbackScheduler();
-      if (fallbackScheduler) {
-        const assignedWorkspaceId = await fallbackScheduler.assignNextWorkspace(
-          userId,
-          maybeSuboptimal,
-        );
-
-        return assignedWorkspaceId;
-      }
-
-      throw new Error("No eligible workspace");
+    // Should always be in oracle mode
+    if (!this.isInOracleMode.getValue()) {
+      throw new Error(
+        "Not in Oracle Mode. If you see this, please contact us. Then try logging out and logging back in.",
```

Comment:
I would maybe take out the "Then try logging out and logging back in". Not a blocking issue, though. If they see this error it's the least of our worries.